src/main/java/duke/tools/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke.tools;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.exceptions.UnrecognisedFormatException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+/**
+ * Handles loading tasks from storage and saving tasks to storage.
+ */
+public class Storage {
+
+    private final String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the storage file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the storage file.
+     *
+     * @return A list of tasks loaded from storage.
+     * @throws IOException If an I/O error occurs.
+     */
+    public List<Task> loadTasksFromStorage() throws IOException {
+        List<Task> tasks = new ArrayList<>();
+        File file = new File(filePath);
+
+        // scan for the storage file
+        try {
+            Scanner s = new Scanner(file);
+            while (s.hasNext()) {
+
+                String curr = s.nextLine();
+                String[] segments = curr.split(" \\| ");
+
+                // check for the correct format - minimum 3 different segments
+                if ((!segments[0].equals("T") && !segments[0].equals("D")
+                        && !segments[0].equals("E")) || segments.length < 3) {
+                    s.close(); // need to close scanner otherwise cannot replace file
+                    throw new UnrecognisedFormatException();
+                }
+
+                boolean isDone = segments[1].equals("1");
```

Comment:
Great work naming your booleans identifiably according to the Java coding standard consistently throughout your code!